pkg/kubelet/sysctl/safe_sysctls.go

```diff
@@ -51,19 +51,31 @@ var safeSysctlsIncludeReservedPorts = []string{
 // - it is namespaced in the container or the pod
 // - it is isolated, i.e. has no influence on any other pod on the same node.
 func SafeSysctlAllowlist() []string {
-	if goruntime.GOOS == "linux" {
-		// make sure we're on a new enough kernel that the ip_local_reserved_ports sysctl is namespaced
-		kernelVersion, err := getKernelVersion()
-		if err != nil {
-			klog.ErrorS(err, "Failed to get kernel version, dropping net.ipv4.ip_local_reserved_ports from safe sysctl list")
-			return safeSysctls
-		}
-		if kernelVersion.LessThan(version.MustParseGeneric(ipLocalReservedPortsMinNamespacedKernelVersion)) {
-			klog.ErrorS(nil, "Kernel version is too old, dropping net.ipv4.ip_local_reserved_ports from safe sysctl list", "kernelVersion", kernelVersion)
-			return safeSysctls
-		}
+	if goruntime.GOOS != "linux" {
+		return nil
 	}
-	return safeSysctlsIncludeReservedPorts
+
+	kernelVersion, err := getKernelVersion()
+	if err != nil {
+		klog.ErrorS(err, "failed to get kernel version, unable to determine which sysctls are available")
+		return safeSysctls
+	}
+
+	var includeSafeSysctlsLimitedByKernelVersion []string
```

Comment:
I'd remove "include" by the name? Not clear why it's there... it's just a list of "safe sysctls limited by kernel version", right?

Reply:
>  it's just a list of "safe sysctls limited by kernel version", right?

No,  the list also includes `safeSysctls `, see https://github.com/kubernetes/kubernetes/pull/118846/files#diff-24ecf51d74242f89b3e4d8eb7ff56a38c3bed5c928bae42e755822996938cf73R77

If we remove 'include' by the name, it looks like we just return "safe sysctls limited by kernel version", but does not include "sysctls that are not limited by kernel version", in fact we returned the sum of the two, right?

Reply:
How about the latest changes?

Reply:
/cc @danwinship Can you please take a look?